[playwright] LPD-54158 Fix playwright test failures in Content Management modules due to LPD-51577 pagination update

boton opened the pull request:
# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-54158

This PR addresses test failures in Content Management Playwright tests that started happening after the pagination logic was updated in [LPD-51577](https://liferay.atlassian.net/browse/LPD-51577). 

# How am I fixing it?

I’ve updated the pagination utility to restrict the allowed delta values to those supported by the portal. The portal now only supports a fixed set of delta values: `5, 10, 20, 40, 60`.
This prevents future inconsistencies or silent failures when using unsupported deltas.

>[!CAUTION]
> However, this change makes the tests significantly more expensive to run, as we now need to create more items in order to trigger the pagination.
> To try to reduce the cost, I attempted to override the default deltas in the search container configuration using these properties:
> - `modules/test/playwright/env/portal-ext.properties`
> - `modules/test/playwright/tests/message-boards-web/env/portal-ext.properties`
>
> **Unfortunately, they seem to have no effect.**

# How can you verify that it works?

Run the playwright affected tests.

```sh
npm run playwright -- test -g "@LPD-52628|@LPD-19384|@LPD-37727|@LPD-39570"
```

# Tested locally

### ✅ Sometimes it passes.
![image](https://github.com/user-attachments/assets/807470cb-7139-40a8-b313-feb39044abca)

### ❌ Other times it still fails due to variability in the number of created items and timing issues.
![image](https://github.com/user-attachments/assets/bb356581-db9a-420f-a879-484fa1b9000d)

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-31.liferay.com/job/test-portal-source-format/7082/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-31.liferay.com/userContent/jobs/test-portal-source-format/builds/7082/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<p><code>relevant</code> and <code>stable</code> testing have been bypassed because all modified files were excluded by the <b>modified.files.global.excludes</b> property in <code>test.properties</code>. The <code>ci:test:relevant</code> and <code>ci:test:stable</code> labels will match the <code>ci:test:sf</code> result.</p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 4 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/13ba7bf2b9b0816bff0efcf168ce65c0d66f4f01">13ba7bf2b9b0816bff0efcf168ce65c0d66f4f01</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/boton/liferay-portal/tree/LPD-54158_TEST-PW_paginator">LPD-54158_TEST-PW_paginator</a><br/>Branch GIT ID: <a href="https://github.com/boton/liferay-portal/commit/4c9757c19e0b95126e60719472a3f6f9cbfb607f">4c9757c19e0b95126e60719472a3f6f9cbfb607f</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-31.liferay.com/job/test-portal-source-format/7082/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-31.liferay.com/userContent/jobs/test-portal-source-format/builds/7082/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-31.liferay.com/job/test-portal-source-format/7082/">test-portal-source-format#7082</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-31.liferay.com/userContent/jobs/test-portal-source-format/builds/7082/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-content-management/liferay-portal/pull/6593">liferay-content-management#6593</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357/build/205112801">[master] ci:test:sf - boton &gt; liferay-content-management - PR#6593 - 2025-04-29[06:17:21]</a><br/></div><div><strong>Testray Build ID:</strong><span></span><code>205112801</code><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-31.liferay.com/job/publish-testray-report/27856/">publish-testray-report#27856</a><br/></div></div>

ambrinchaudhary reviewed (approved):
Fixes LGTM. Let's wait to see if you receive comments on the slack channel to avoid generating that much entries. If not possible, we can move forward this pull.

boton commented:
Rebased and removed the 5 and 10 options from the paginator.
Tests are more or less working, some are still failing due to the large number of items we need to create.
We could improve this in a separate task by using a batch endpoint (though I'm not sure how to implement that yet).

![image](https://github.com/user-attachments/assets/94a96d2f-c064-4059-a151-c3d0e8f20f61)

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>content-management-playwright</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest(master)/6809/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-35.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/6809/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:x: ci:test:content-management-playwright - 11 out of 17 jobs passed in 1 hour 24 minutes</h3><details><summary>Click here for more details.</summary><br/><p>This pull is eligible for <a href="https://grow.liferay.com/share/CI+liferay-continuous-integration+GitHub+Commands#General-Commands">reevaluation</a>. When this <a href="https://test-1-40.liferay.com/job/test-portal-acceptance-upstream-dxp(master)/1345">upstream build</a> has completed, using the following CI command will compare this pull request result against a more recent upstream result:<pre>ci:reevaluate:1357401_6809</pre></p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/b08e35cf6a67ed0527d8614093839e770118d34a">b08e35cf6a67ed0527d8614093839e770118d34a</a></p><h4>Upstream Comparison:</h4><p>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/e82707c5af70f74ee11f5c3c551026af92a12e9f">e82707c5af70f74ee11f5c3c551026af92a12e9f</a><br/>Jenkins Build URL: <a href="https://testray.liferay.com/#/project/35392/routines/590307/build/209951404">EE Development Acceptance (master) - 1344 - 2025-05-06[00:33:34]</a></p><details><summary><strong>ci:test:content-management-playwright - 11 out of 17 jobs PASSED</strong></summary><div><h4>6 Failed Jobs:</h4><ul><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest(master)/6809/"><strike>test-portal-acceptance-pullrequest(master)</strike></a></li><li>playwright-js-tomcat90-postgresql163<ul><li><a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1399500/">playwright-js-tomcat90-postgresql163/0/0</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1259942/"><strike>playwright-js-tomcat90-postgresql163/1/0</strike></a></li><li><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1341023/">playwright-js-tomcat90-postgresql163/2/0</a></li><li><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1341025/">playwright-js-tomcat90-postgresql163/3/0</a></li><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1233473/">playwright-js-tomcat90-postgresql163/4/0</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/816129/">playwright-js-tomcat90-postgresql163/5/0</a></li><li><a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1399503/"><strike>playwright-js-tomcat90-postgresql163/6/0</strike></a></li><li><a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1399501/"><strike>playwright-js-tomcat90-postgresql163/7/0</strike></a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/816128/"><strike>playwright-js-tomcat90-postgresql163/8/0</strike></a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1259943/">playwright-js-tomcat90-postgresql163/9/0</a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1167633/">playwright-js-tomcat90-postgresql163/10/0</a></li><li><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1341024/">playwright-js-tomcat90-postgresql163/11/0</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1259944/">playwright-js-tomcat90-postgresql163/12/0</a></li><li><a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1399502/"><strike>playwright-js-tomcat90-postgresql163/13/0</strike></a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1259941/">playwright-js-tomcat90-postgresql163/14/0</a></li><li><a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1399499/">playwright-js-tomcat90-postgresql163/15/0</a></li></ul></li></ul></div><details><summary><strong>11 Successful Jobs:</strong></summary><ul/></details></details><h5>For more details click <a href="https://test-1-35.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/6809/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1259942//consoleText">playwright-js-tomcat90-postgresql163/1/0</a> - <a href="https://storage.cloud.google.com/testray-results/2025-05/test-1-35/test-portal-acceptance-pullrequest(master)/6809/playwright-js-tomcat90-postgresql163/1/0/playwright-report/index.html">Playwright Report</a><ol><li><details><summary><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1259942/testReport/asset-tags-admin-web_main_tags.spec/ts">asset-tags-admin-web/main/tags.spec.ts</a></summary><div><ol><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1259942//testReport/asset-tags-admin-web_main_tags.spec/ts/Accessing_via_keyboard_navigation_the_tooltip_of_the_back_button_is_Go_to_Tags_when_adding_or_merging_Tags">Accessing via keyboard navigation the tooltip of the back button is Go to Tags when adding or merging Tags</a></li></ol></div></details></li></ol></div></li><li><div><a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1399502//consoleText">playwright-js-tomcat90-postgresql163/13/0</a> - <a href="https://storage.cloud.google.com/testray-results/2025-05/test-1-35/test-portal-acceptance-pullrequest(master)/6809/playwright-js-tomcat90-postgresql163/13/0/playwright-report/index.html">Playwright Report</a><ol><li><details><summary><a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1399502/testReport/message-boards-web_main_message-boards-web.spec/ts">message-boards-web/main/message-boards-web.spec.ts</a></summary><div><ol><li><a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1399502//testReport/message-boards-web_main_message-boards-web.spec/ts/Do_not_show_site_in_breadcrumb">Do not show site in breadcrumb</a></li></ol></div></details></li></ol></div></li></ol><hr/><details><summary><strong>Failures in common with <a href="https://test-1-40.liferay.com/job/test-portal-acceptance-upstream-dxp(master)">acceptance upstream results</a> at e82707c5af70f74ee11f5c3c551026af92a12e9f:</strong></summary><ol><li><div><a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1399503//consoleText">playwright-js-tomcat90-postgresql163/6/0</a> - <a href="https://storage.cloud.google.com/testray-results/2025-05/test-1-35/test-portal-acceptance-pullrequest(master)/6809/playwright-js-tomcat90-postgresql163/6/0/playwright-report/index.html">Playwright Report</a><ol><li><details><summary><a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1399503/testReport/announcements-web_main_announcements-web.spec/ts">announcements-web/main/announcements-web.spec.ts</a></summary><div><ol><li><a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1399503//testReport/announcements-web_main_announcements-web.spec/ts/The_distribution_scopes_of_the_Announcements_widget_can_be_configured">The distribution scopes of the Announcements widget can be configured</a></li></ol></div></details></li></ol></div></li><li><div><a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1399501//consoleText">playwright-js-tomcat90-postgresql163/7/0</a> - <a href="https://storage.cloud.google.com/testray-results/2025-05/test-1-35/test-portal-acceptance-pullrequest(master)/6809/playwright-js-tomcat90-postgresql163/7/0/playwright-report/index.html">Playwright Report</a><ol><li><details><summary><a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1399501/testReport/knowledge-base-web_main_knowledge-base-web.spec/ts">knowledge-base-web/main/knowledge-base-web.spec.ts</a></summary><div><ol><li><a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1399501//testReport/knowledge-base-web_main_knowledge-base-web.spec/ts/LPD_27537__Article_should_be_shown_to_guest_users">LPD-27537: Article should be shown to guest users</a></li></ol></div></details></li></ol></div></li><li><div><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/816128//consoleText">playwright-js-tomcat90-postgresql163/8/0</a> - <a href="https://storage.cloud.google.com/testray-results/2025-05/test-1-35/test-portal-acceptance-pullrequest(master)/6809/playwright-js-tomcat90-postgresql163/8/0/playwright-report/index.html">Playwright Report</a><ol><li><details><summary><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/816128/testReport/document-library-web_main_dmAICreator.spec/ts">document-library-web/main/dmAICreator.spec.ts</a></summary><div><ol><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/816128//testReport/document-library-web_main_dmAICreator.spec/ts/Create_AI_Image_option_is_hidden_when_disabled_from_Instance_Settings">Create AI Image option is hidden when disabled from Instance Settings</a></li></ol></div></details></li></ol></div></li></ol></details></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest(master)/6809/">test-portal-acceptance-pullrequest(master)#6809</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-35.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/6809/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>content-management-playwright<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-content-management/liferay-portal/pull/6593">liferay-content-management#6593</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357/build/210211835">[master] ci:test:content-management-playwright - boton &gt; liferay-content-management - PR#6593 - 2025-05-06[10:10:45]</a><br/></div><div><strong>Testray Build ID:</strong><span></span><code>210211835</code><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-35.liferay.com/job/publish-testray-report/14932/">publish-testray-report#14932</a><br/></div></div>

boton commented:
failures looks unrelated 🥳

boton commented:
ci:forward

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**stable**

The pull request will automatically be forwarded to the user `brianchandotcom` If the following test suites pass:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**stable**

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>relevant</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest(master)/11249/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-29.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/11249/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<h1>Build completed.</h1><p>Jenkins URL: <a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest(master)/11249/">test-portal-acceptance-pullrequest(master)</a></p><p>The <code>relevant</code> test suite did not execute tests because all modified files were excluded by the <b>modified.files.global.excludes</b> property in <code>test.properties</code>.</p><p>Please use <code>ci:test:sf</code> and/or <code>ci:forward</code> for this pull request to be forwarded or eligible for forwarding.</p>

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-9.liferay.com/job/test-portal-source-format/6637/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-9.liferay.com/userContent/jobs/test-portal-source-format/builds/6637/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<p><code>relevant</code> and <code>stable</code> testing have been bypassed because all modified files were excluded by the <b>modified.files.global.excludes</b> property in <code>test.properties</code>. The <code>ci:test:relevant</code> and <code>ci:test:stable</code> labels will match the <code>ci:test:sf</code> result.</p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 6 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/387d567711c76d628a32704dffacf46e974102ce">387d567711c76d628a32704dffacf46e974102ce</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/boton/liferay-portal/tree/LPD-54158_TEST-PW_paginator">LPD-54158_TEST-PW_paginator</a><br/>Branch GIT ID: <a href="https://github.com/boton/liferay-portal/commit/e363b128e596a46b422d17693a39188aac9e0a80">e363b128e596a46b422d17693a39188aac9e0a80</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-9.liferay.com/job/test-portal-source-format/6637/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-9.liferay.com/userContent/jobs/test-portal-source-format/builds/6637/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
Pull request has been successfully forwarded to  https://github.com/brianchandotcom/liferay-portal/pull/161588
[Console](https://test-1-9.liferay.com/job/test-portal-source-format/6637//consoleFull)

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-9.liferay.com/job/test-portal-source-format/6637/">test-portal-source-format#6637</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-9.liferay.com/userContent/jobs/test-portal-source-format/builds/6637/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-content-management/liferay-portal/pull/6593">liferay-content-management#6593</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357/build/210884409">[master] ci:test:sf - boton &gt; liferay-content-management - PR#6593 - 2025-05-07[03:35:03]</a><br/></div><div><strong>Testray Build ID:</strong><span></span><code>210884409</code><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-9.liferay.com/job/publish-testray-report/7420/">publish-testray-report#7420</a><br/></div></div>
